man/systemd.netdev.xml

```diff
@@ -1979,6 +1982,77 @@
     </variablelist>
   </refsect1>
 
+  <refsect1>
+    <title>[Batadv] Section Options</title>
+    <para>The [Batadv] section only applies for
+    netdevs of kind <literal>batadv</literal> and accepts the
+    following keys:</para>
+
+    <variablelist class='network-directives'>
+      <varlistentry>
+        <term><varname>GatewayMode=</varname></term>
+        <listitem>
+          <para>The GatewayMode is a string of either "off", "server" or "client"</para>
+        </listitem>
+      </varlistentry>
+      <varlistentry>
+        <term><varname>BridgeLoopAvoidance=</varname></term>
+        <listitem>
+          <para>Enables (1) or disables (0) BridgeLoopAvoidance</para>
+        </listitem>
+      </varlistentry>
+      <varlistentry>
+        <term><varname>DistributedArpTable=</varname></term>
+        <listitem>
+          <para>Enables (1) or disables (0) DistributedArpTable</para>
+        </listitem>
+      </varlistentry>
+      <varlistentry>
+        <term><varname>Fragmentation=</varname></term>
+        <listitem>
+          <para>Enables (1) or disables (0) Fragmentation</para>
+        </listitem>
+      </varlistentry>
+      <varlistentry>
+        <term><varname>HopPenalty=</varname></term>
+        <listitem>
+          <para>The hop penalty setting allows to modify batman-adv's preference for multihop
+          routes vs. short routes. The value is applied to the TQ of each forwarded OGM, thereby
+          propagating the cost of an extra hop (the packet has to be received and retransmitted
+          which costs airtime). A higher hop penalty will make it more unlikely that other nodes
+          will choose this node as intermediate hop towards any given destination.</para>
+        </listitem>
+      </varlistentry>
+      <varlistentry>
+        <term><varname>OrigInterval=</varname></term>
+        <listitem>
+          <para>The value specifies the interval (milliseconds) in which batman-adv floods the
+          network with its protocol information.</para>
+        </listitem>
+      </varlistentry>
+      <varlistentry>
+        <term><varname>GatewayBandwithDown=</varname></term>
```

Comment:
Please use the generic parser for this (`cofnig_parse_iec_size()` or similar) and use the usual description:
> If the value is suffixed with K, M, G or T, the specified memory size is parsed as Kilobytes, Megabytes, Gigabytes, or            Terabytes (with the base 1024), respectively.